pkg/cloudprovider/vsphereparavirtual/apis/nsxnetworking/v1alpha1/routetable_types.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RouteTable describe a routing table resource.
+type RouteTable struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   RouteTableSpec   `json:"spec"`
+	Status RouteTableStatus `json:"status,omitempty"`
+}
+
+// RouteTableSpec defines the desired state of RouteTable.
+type RouteTableSpec struct {
+	// Routes is the list of desired routes.
+	Routes []Route `json:"routes"`
+}
+
+// Route defines a route entry for traffic.
+type Route struct {
+	// Name defines the name of this route.
```

Comment:
@dantingl I think maybe we need to add annotations/ownerRef here to identify the cluster name and node name extracting clusterName and nodeName from Route Name might cause errors because clusterName and nodeName can have different formats, e.g. including `-` character
so if route name is: `node-name-0.0.0.0-0-cluster-name`, it's hard to extract clusterName and nodeName directly from it